File: mobile/ios/RAHATOMIR/PrayerWidgetModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PrayerWidget, NSObject)

RCT_EXTERN_METHOD(setPayload:(NSString *)json)
RCT_EXTERN_METHOD(setQiblaHeading:(nonnull NSNumber *)heading)

@end

File: mobile/ios/RAHATOMIR/PrayerWidgetModule.swift
import Foundation
import React

@objc(PrayerWidget)
class PrayerWidgetModule: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func setPayload(_ json: String) {
    PrayerWidgetStore.savePayload(json)
  }

  @objc func setQiblaHeading(_ heading: NSNumber) {
    PrayerWidgetStore.saveQiblaHeading(heading.doubleValue)
  }
}
